packages/hash/api/src/collab/Instance.ts

```diff
@@ -71,6 +74,28 @@ type ActionUpdate = {
 
 type Update = StepUpdate | StoreUpdate | ActionUpdate;
 
+/**
+ * Returns a BlockFilter for a specified block, but is guaranteed to return the
+ * same object given the same block account id and entity id – allows using
+ * in Map and Set
+ */
+const blockFilter = (() => {
```

Comment:
Optional suggestions:

- We can start this function name with a verb and thus help unprepared readers. Is it `getBlockFiler` / `findBlockFilter` / `generateBlockFilter` etc.? Various function names communicate different invocation costs, which can be handy for spotting inefficiencies and mistakes.

- Not sure we benefit much from putting `const filters = new Map<string, BlockFilter>();` into a closure. It looks smart, but is not that easy to understand, especially for junior developers. If we can avoid extra wrapping, we can add explicit typings to the function. It will be easier for others (and you in six months) to maintain this file. Explicit return types make TS a bit faster and also move linting problems closer to their source.